src/ticketmap/index.tsx

```diff
@@ -61,6 +69,28 @@ interface TicketGroupsBySectionByZone {
   }
 }
 
+const createDeepEqualSelector = createSelectorCreator(defaultMemoize, isEqual)
+
+const $missingSectionIds = createDeepEqualSelector(
+  (state: State) => state.ticketGroups,
+  (state: State) => state.sectionZoneMapping,
```

Comment:
Do we need to pass the whole state in and have these functions which pop off top level properties? Could we create a selector, and maintain its memoization functionality, by removing these first two functions and calling the selector like `$missingSectionIds(this.state.ticketGroups, this.state.sectionZoneMapping)`?

Reply:
Allowing selectors to access the entire state has a ton of benefits:
- callers don't need to know the shape of the state
- callers don't need to pass `undefined` values if certain selector inputs aren't available
- selectors can perform complex logic in these extraction functions
```
const $missingSectionIds = createDeepEqualSelector(
  (state: State) => state.ticketGroups || state.defaultTicketGroups
  /* ... */
)
```
- the extraction functions could be replaced by other selectors and callers don't need to be updated
- selector inputs can change without callers needing to be updated